### DISCUSSION TITLE
my own ideas for enemy types

### DISCUSSION THREAD

**Discussion by MW2917** [efroemling/ballistica#662] [Show and tell]
**MAX bots (crafted)**
Mainly cuz I wanted to see a strong enemy with silver and gold theme. 
![BombSquad 09_02_2024 10_15_19](https://github.com/efroemling/ballistica/assets/160723853/00ec99e1-e2d5-459c-a0b0-e16028c0f861)
![BombSquad 09_02_2024 09_20_10](https://github.com/efroemling/ballistica/assets/160723853/b69dab3e-2054-4dd3-83eb-04af4b746a55)
The sliver/gold bots were the once I maxed out. But they seemed to only do slightly better than PRO bots due to PROs already have very high aggression.
**TNT bots (cradted)**
The idea is to have bots enhancing other bombers' attack power/range by planting TNTs. And players can also use their attacks.

https://github.com/efroemling/ballistica/assets/160723853/813fe2a1-a46c-4324-821a-0a4e7b3d6c8d

The one that is using Mel is a type that brings the danger to you like sticky bot does, filling your surroundings with TNT, and gladly drags you to hell alongside himself.
The other one using Bernard is the turret, meant to deploy TNTs from a distance alongside normal bombs, to compliment the attacks.
But the weight of the TNT made it impossible for ranged attacks. And it's easier to hit a square TNT, causing them to be too easy to defeat. So I made a round, compact TNT pack meant to be thrown just for this bot. (Could use it further tho)

https://github.com/efroemling/ballistica/assets/160723853/a3708bd8-7998-4da3-ba34-3e15211cde65

**Rabbit bomb (crafted)**
I saw it once in an online party, where a power-up gives you a 2 egg pack, when deployed, it'll start moving until it successfully hatches a rabbit that will explode, or attack it to make it instantly hatch.
It's merely a matter of adding bouncy into an explodey bot and turn it into a bunny, or make a bouncy bot that will blow up, and I did the one behind. This bot is really meant to be summoned, which leads me to the next bot I wanna talk about.

**bot that summons other monsters (bots) (still idea)**
I saw it in the bots vs bots (unreleased version) where a wizard would deploy skulls, and after impact, incomes a skeleton.
Not sure if I should class a new bomb or define it in bot class, but if it's done the first way, that would mean players can possibly do so too, although I would say if it's gonna be a new power-up, it'll only work in multiplayers, as the bot spawned will not be like the once in bunnybuddy, it'll just attack anyone it see, including the summoner.

**Freeze puncher (also not real yet)(not on 1.7/8 anyway)**
Also seen in the bots vs bots mod, but an older, released one this time (from 1.4). So the matter will be to port the code that defines it's actions into api8. 
Basically, it's a penguin that can freeze anyone it punches, so should be a tough enemy since you can't receive a small injury or you'll die of hypothermia.

_And that's my ideas for bots. Pls tell me if it's good or sh-_
_thx_
